cmd/context.go

```diff
@@ -169,17 +169,27 @@ func showContext(client *client.Client, vcsType, orgName, contextName string) er
 	return nil
 }
 
-func readSecretValue() (string, error) {
+// ReadSecretValue reads a secret from a buffer
+func ReadSecretValue() (string, error) {
 	stat, _ := os.Stdin.Stat()
+
+	buffSize, err := os.Stdin.Stat()
+	if err != nil {
+		return "", err
+	}
+
+	reader := bufio.NewReaderSize(os.Stdin, int(buffSize.Size()))
+
 	if (stat.Mode() & os.ModeCharDevice) == 0 {
-		bytes, err := ioutil.ReadAll(os.Stdin)
+		bytes := make([]byte, buffSize.Size())
+		_, err := io.ReadFull(reader, bytes)
```

Comment:
Did you happen to experience errors other than `failed to read input: bufio.Scanner: token too long` from this code path?

I believe that `ioutil.ReadAll` will read the entire contents of the provided `io.Reader` without any limitations. I believe the ['512'](https://golang.org/src/io/io.go?s=19249:19287#L616) refers to the initial buffer size, which will dynamically grow until EOF/OOM occurs, so I'm not certain changing this to `io.ReadFull` will address that error specifically.